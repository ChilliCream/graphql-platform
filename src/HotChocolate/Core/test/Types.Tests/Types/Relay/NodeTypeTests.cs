using HotChocolate.Types.Relay;
using Xunit;

namespace HotChocolate.Types
{
    public class NodeTypeTests
        : TypeTestBase
    {
        [Fact]
        public void InitializeExplicitFieldWithImplicitResolver()
        {
            // arrange
            // act
            NodeType nodeInterface = CreateType(
                new NodeType(),
                b => b.ModifyOptions(o => o.StrictValidation = false));

            // assert
            Assert.Equal(
                "Node",
                nodeInterface.Name);

            Assert.Equal(
                "The node interface is implemented by entities that have " +
                "a global unique identifier.",
                nodeInterface.Description);

            Assert.Collection(nodeInterface.Fields,
                t =>
                {
                    Assert.Equal("id", t.Name);
                    Assert.IsType<IdType>(
                        Assert.IsType<NonNullType>(t.Type).Type);
                });
        }
    }
}
