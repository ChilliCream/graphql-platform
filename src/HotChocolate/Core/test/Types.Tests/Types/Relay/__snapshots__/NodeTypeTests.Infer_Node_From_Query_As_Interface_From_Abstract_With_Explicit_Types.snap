schema {
  query: Query12
}

interface Document implements Node {
  id: ID!
}

"The node interface is implemented by entities that have a global unique identifier."
interface Node {
  id: ID!
}

type DrivingLicense implements Node & Document {
  id: ID!
}

type Query12 {
  "Fetches an object given its ID."
  node("ID of the object." id: ID!): Node
  "Lookup nodes by a list of IDs."
  nodes("The list of node IDs." ids: [ID!]!): [Node]!
  documentById(id: ID!): Document
}
